### PR TITLE
Fixes the duplicate doc comment display in case of union property

### PR DESCRIPTION
**Pull request by sheetalkamat**
Handles #1237

### CODE REVIEW

**Comment by mhegazy** [comment]
:+1:

File: src/services/services.ts [hunk 1]
```diff
@@ -335,36 +335,43 @@ module ts {
...
-                if (canUseParsedParamTagComments && declaration.kind === SyntaxKind.Parameter) {
-                    ts.forEach(getJsDocCommentTextRange(declaration.parent, sourceFileOfDeclaration), jsDocCommentTextRange => {
-                        var cleanedParamJsDocComment = getCleanedParamJsDocComment(jsDocCommentTextRange.pos, jsDocCommentTextRange.end, sourceFileOfDeclaration);
-                        if (cleanedParamJsDocComment) {
-                            jsDocCommentParts.push.apply(jsDocCommentParts, cleanedParamJsDocComment);
-                        }
-                    });
-                }
+            ts.forEach(declarations,(declaration, indexOfDeclaration) => {
+                if (indexOf(declarations, declaration) === indexOfDeclaration) {
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: I don't understand, equality works by reference, when would this not be the case?

File: src/services/services.ts [hunk 2]
```diff
@@ -335,36 +335,43 @@ module ts {
...
-                // If it is parameter - try and get the jsDoc comment with @param tag from function declaration's jsDoc comments
-                if (canUseParsedParamTagComments && declaration.kind === SyntaxKind.Parameter) {
-                    ts.forEach(getJsDocCommentTextRange(declaration.parent, sourceFileOfDeclaration), jsDocCommentTextRange => {
-                        var cleanedParamJsDocComment = getCleanedParamJsDocComment(jsDocCommentTextRange.pos, jsDocCommentTextRange.end, sourceFileOfDeclaration);
-                        if (cleanedParamJsDocComment) {
-                            jsDocCommentParts.push.apply(jsDocCommentParts, cleanedParamJsDocComment);
-                        }
-                    });
-                }
+            ts.forEach(declarations,(declaration, indexOfDeclaration) => {
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: Space after comma, consider just making this a for loop.

File: src/services/services.ts [hunk 3]
```diff
@@ -335,36 +335,43 @@ module ts {
...
+                if (indexOf(declarations, declaration) === indexOfDeclaration) {
+                    // Already collected the doc comments from this declaration.
+                    // Eg. var a: Array<string> | Array<number>; a.length
+                    // The property length will have two declarations of property length coming 
+                    // from Array<T> - Array<string> and Array<number>
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: Move this comment above the `if`, explain that the two declarations will *appear* to be the same declaration here even though they are from different instantiations of the same type.

**Pull request merged**